### ISSUE TITLE
LibRedirect integration not working (?): use Farside instead

### ISSUE THREAD

**Issue by ghost** [LinkSheet/LinkSheet#175]
### Feature description

Okay, I've been using the app and I find it very interesting; but don't know if I got it right, this implementation only works if you have the extension installed in your browser?
I tried to access Nitter through the app using something like "twitter.com/x" and not being redirected.
As I don't use the Nightly version of _Firefox for Android_, I'm unable to install the extension.

### Farside
My suggestion is to use [Farside](https://farside.link/) instead for redirection, since the LibRedirect integration is still experimental and unfinished. Could you think about it? Thanks.

**Comment by 1fexd** [comment]
> Okay, I've been using the app and I find it very interesting; but don't know if I got it right, this implementation only works if you have the extension installed in your browser?

No, links are automatically rewritten within LinkSheet, it does not require the browser extension.

> I tried to access Nitter through the app using something like "twitter.com/x" and not being redirected.
 
Looks like a bug. Will check it out.

> My suggestion is to use [Farside](https://farside.link/) instead for redirection

Never heard of this service, but looks very interesting. Maybe I am going to add it as an alternative to LibRedirect.

> since the LibRedirect integration is still experimental and unfinished.

The reason I stated it is experimental/unfinished is that some features may not be completely done, but it should work fine mostly.

**Comment by zakius** [comment]
some links aren't handled directly by libredirect rules and have to get redirected first, did you try it with redirects enabled?

**Comment by 1fexd** [comment]
> EDIT: but it seems the LibRedirect integration hangs quite often on different links making it impossible to open them at all

Could you provide some of those links? Have you tested them on the desktop extension?

**Comment by zakius** [comment]
it looks like the issue is more complex, once I uninstalled RS File Manager (as I don't need it anymore and it popped up as a possible handler for all links) it doesn't hang anymore, but when I had it installed it did hang only with libredirect enabled

or it's something even more convoluted and I don't know how to reproduce it anymore, sorry

**Comment by r1klz** [comment]
> Never heard of this service, but looks very interesting. Maybe I am going to add it as an alternative to LibRedirect.

Glad to hear! Yeah, I heard about this service viewing recent issues from Piped and found it useful. But, per example, it sorts by online servers (the website works by itself), except that it doesn't filter by the proxy server that Piped uses to load videos; and if the proxy isn't working (as now in some instances), it wont load videos. Despite that, it's nice to see which instances are working.

> Looks like a bug. Will check it out.

Oh, and could you reproduce?

> The reason I stated it is experimental/unfinished is that some features may not be completely done, but it should work fine mostly.

I see. I don't want you to stop your project because that "bug". My suggestion was only for a momentary solution. LibRedirect is a more mature project, supporting more services than Farside. 👍

**Comment by 1fexd** [comment]
The thing about LibRedirect is, that LinkSheet's implementation is almost always out-dated, as new updates (to the instance list and to add new services) are only added in new LinkSheet releases. There currently is no update mechanism (but it is already tracked in #142), meaning LinkSheet may still list some instances which are down/don't exist anymore. Also, if some code changes in the browser extension which rewrites links to free-frontends, I have to change it in (https://github.com/1fexd/libredirectkt)[libredirectkt] (the library LinkSheet uses to offer LibRedirect integration) as well. This is a huge pain in the butt as the browser extension is written in Javascript while my library is written in Kotlin, meaning I have to rebuild what they have already built in Javascript, but even worse, there appear to be no tests (I just opened an issue on their issue tracker, asking if they have any), so I can not even test if my implementation works correctly. 

After having a look at Farside, I am probably going to retire the LibRedirect integration and replace it with Farside in the next release. If you really depend on a feature Farside does not support, I suggest you open an feature request on their repository, or use the LibRedirect browser extension in Firefox by following [this tutorial](https://www.androidpolice.com/install-add-on-extension-mozilla-firefox-android/) (you do not need Firefox Beta, it works on Fennec F-Droid as well; Also, Mozilla [announced Android addon they are finally working on adding propery extension support on mobile the other day](https://blog.mozilla.org/addons/2023/08/10/prepare-your-firefox-desktop-extension-for-the-upcoming-android-release/), so the workaround will not be needed anymore in the future.
